apps/mark-scan/frontend/src/pages/inserted_preprinted_ballot_screen.tsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+
+import { Button, Icons, P } from '@votingworks/ui';
+import { BallotStyleId, PrecinctId, VotesDict } from '@votingworks/types';
+import { assert } from '@votingworks/basics';
+
+import * as api from '../api';
+import { CenteredCardPageLayout } from '../components/centered_card_page_layout';
+
+export interface InsertedPreprintedBallotScreenProps {
+  activateCardlessVoterSession: (
+    precinctId: PrecinctId,
+    ballotStyleId: BallotStyleId
+  ) => void;
+  setVotes: (votes: VotesDict) => void;
+}
+
+export function InsertedPreprintedBallotScreen(
+  props: InsertedPreprintedBallotScreenProps
+): React.ReactNode {
+  const { activateCardlessVoterSession, setVotes } = props;
+
+  const startSessionWithPreprintedBallot =
+    api.startSessionWithPreprintedBallot.useMutation().mutate;
+
+  const returnPreprintedBallot =
+    api.returnPreprintedBallot.useMutation().mutate;
+
+  const interpretationQuery = api.getInterpretation.useQuery();
+  const interpretation = interpretationQuery.data;
+
+  const onPressReview = React.useCallback(() => {
+    assert(interpretation);
+    assert(interpretation.type === 'InterpretedBmdPage');
+
+    const { ballotStyleId, precinctId } = interpretation.metadata;
+
+    activateCardlessVoterSession(precinctId, ballotStyleId);
+    startSessionWithPreprintedBallot();
```

Comment:
It looks like the state machine can end cardless voter auth [here](https://github.com/votingworks/vxsuite/blob/main/apps/mark-scan/backend/src/custom-paper-handler/state_machine.ts#L619). Would it be cleaner to make the state machine responsible for all cardless voter auth changes in this flow?

Reply:
Not a hardened opinion, but leaning toward this path in the short term -- the UI is current the only place where we activate sessions (granted, because it depends on information from users). So, this feels consistent with that, but would be open to larger change to move that logic as well.

I think even if we moved the `activateCardlessVoterSession` call to the state machine (assuming that's what were referring to), it would still make sense to keep the choice of `precinctId` and `ballotStyleId` here in the UI, at which point, the only change we'd be making would be to move the logic of calling out to the auth library from the backend API into the state machine (and my long-term bias is to separate the concerns of auth state, UI state, and machine state anyway, to make things easier to reason about).

Reply:
That sounds good - thanks for laying out the reasoning